lib/fdo.c

```diff
@@ -121,7 +121,10 @@ fdo_sdk_status fdo_sdk_run(void)
 			++error_count;
 			if (error_count == ERROR_RETRY_COUNT) {
 				LOG(LOG_INFO, "*********Retry(s) done*********\n");
-				g_fdo_data->state_fn = &_STATE_Shutdown;
+				g_fdo_data->state_fn = &_STATE_Shutdown_Error;
```

Comment:
Why not exit from this condition?

Reply:
_STATE_Shutdown_Error is basically a wrapper over _STATE_SHUTDOWN_, that returns false, representing that we are clearing out some previously allocated memory due to an error.
In case of TO2 success, _STATE_Shutdown is used to do clean-up, returning true.